### PR TITLE
Feat(eos_cli_config_gen): Add schema for event_handlers

### PR DESCRIPTION
**Pull request by KonikaChaurasiya-GSLab**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/<data_model_key>.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/<>.j2` and `templates/documentation/<>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

Note: Also updated molecule scenarios `eos_cli_config_gen` & `eos_cli_config_gen_v4.0` to conform with what is stated we support in the documentation for `event_handlers.[].trigger`

### Reviewer Checklist

- Reviewer 1: Claus
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [ ] Verify that CI pass

- Reviewer 2: Emil
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/event_handlers.schema.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  event_handlers:
+    type: list
+    primary_key: name
+    convert_types:
+    - dict
+    display_name: Event Handler
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/event_handlers.schema.yml]:
```suggestion
    display_name: Event Handlers
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/event_handlers.schema.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            Configure event trigger condition. Only supports on-logging
+        regex:
+          type: str
+          description: |
+            Regular expression to use for searching log messages. Required for on-logging trigger
+        asynchronous:
+          type: bool
+          default: False
+          description: |
+            Set the action to be non-blocking. if unset, default is False
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/event_handlers.schema.yml]: The default column captures this information, IMO we don't need to repeat it here!
```suggestion
            Set the action to be non-blocking.
```

**Review by carlbuchmann — APPROVED** [review]
LGTM - an optional enhancement to the description has been suggested!

**Review by carlbuchmann — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md [hunk 3]
```diff
@@ -87,6 +87,34 @@ daemons:
...
+### Variables
+
+| Variable | Type | Required | Default | Value Restrictions | Description |
+| -------- | ---- | -------- | ------- | ------------------ | ----------- |
+| [<samp>event_handlers</samp>](## "event_handlers") | List, items: Dictionary |  |  |  | Event Handlers |
+| [<samp>&nbsp;&nbsp;- name</samp>](## "event_handlers.[].name") | String |  |  |  | Event Handler Name |
+| [<samp>&nbsp;&nbsp;&nbsp;&nbsp;action_type</samp>](## "event_handlers.[].action_type") | String |  |  | Valid Values:<br>- bash<br>- increment<br>- log | Action Type<br>Type of action<br> |
+| [<samp>&nbsp;&nbsp;&nbsp;&nbsp;action</samp>](## "event_handlers.[].action") | String |  |  |  | Command to execute<br> |
+| [<samp>&nbsp;&nbsp;&nbsp;&nbsp;delay</samp>](## "event_handlers.[].delay") | Integer |  |  |  | Event-handler delay in seconds<br> |
+| [<samp>&nbsp;&nbsp;&nbsp;&nbsp;trigger</samp>](## "event_handlers.[].trigger") | String |  |  |  | Configure event trigger condition. Only supports on-logging<br> |
```
Comment (emilarista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md]: Shouldn't trigger also have a value restriction here if we only support on-logging?

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md]: Is that true, though? Why do we have such statement in current documentation "Only supports on-logging"?
The template itself accepts any value.
Reading the documentation, I don't see why we wouldn't support `on-boot` for example...
https://www.arista.com/en/um-eos/eos-command-line-interface-cli?searchword=event-handler

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md]: In discussion with @ClausHolbechArista , we will adhere to the current documentation, and I will add this check.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by emilarista — APPROVED** [review]
LGTM!

**Pull request merged**